test/lib/mock-encryptor.js

```diff
@@ -1,19 +1,64 @@
 const sinon = require('sinon');
 
-const mockHex = '0xabcdef0123456789';
-const mockKey = Buffer.alloc(32);
+const PASSWORD = 'password123';
+const MOCK_ENCRYPTION_KEY =
+  '{"alg":"A256GCM","ext":true,"k":"wYmxkxOOFBDP6F6VuuYFcRt_Po-tSLFHCWVolsHs4VI","key_ops":["encrypt","decrypt"],"kty":"oct"}';
+const INVALID_PASSWORD_ERROR = 'Incorrect password.';
+
+const MOCK_HEX = '0xabcdef0123456789';
+const MOCK_KEY = Buffer.alloc(32);
 let cacheVal;
 
 module.exports = {
   encrypt: sinon.stub().callsFake(function (_password, dataObj) {
     cacheVal = dataObj;
-    return Promise.resolve(mockHex);
+
+    return Promise.resolve(MOCK_HEX);
   }),
 
-  decrypt(_password, _text) {
+  encryptWithDetail: sinon.stub().callsFake(function (_password, dataObj) {
```

Comment:
It'd be nice to have test coverage for a mobile encryptor as well. Maybe we can leave this as-is, and move these modifications to a new mock encryptor module? Or set two up here, and return them as named exports rather than as the default export.

Then we can ensure things blow up properly if `cacheEncryptionKey` is enabled with an encryptor that doesn't support it.

Reply:
Converting this to a class might make that easier, as was done here: https://github.com/MetaMask/controllers/blob/main/tests/mocks/mockEncryptor.ts

Then we can extend the base encryptor to create the one with the browser-passworder additional methods.